### PR TITLE
fix: re-enable make proto-check and update proto options

### PR DESCRIPTION
**Pull request by MSalopek**
## Description

Closes: #1041 

`make proto-check` was disabled temporarily - this enables it.

`make proto-gen` was creating `/interchain-security/v2/*` stubs - this bumps it to `v3`.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Provided a link to the relevant issue or specification
- [x] Reviewed "Files changed" and left comments if necessary 
- [x] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] Confirmed all author checklist items have been addressed
- [ ] Confirmed that this PR does not change production code

### CODE REVIEW

**Review by mpoke — APPROVED** [review]
LGTM

File: .github/workflows/automated-tests.yml [hunk 1]
```diff
@@ -24,8 +24,8 @@ jobs:
         uses: actions/setup-go@v4
         with:
           go-version: "1.19" # The Go version to download (if necessary) and use.
-      # - name: Proto Check
-      #   run: make proto-check
+      - name: Proto Check
```
Comment (smarshall-spitzbart) [inline .github/workflows/automated-tests.yml:27]: Merge https://github.com/cosmos/interchain-security/pull/1063 into this branch to make this check pass

**Review by smarshall-spitzbart — APPROVED** [review]
Approval once comment is addressed

File: .github/workflows/automated-tests.yml (hunk 1 above)
Comment (MSalopek) [reply .github/workflows/automated-tests.yml:27]: Merged

**Pull request merged**